### PR TITLE
WIP: Update1.x

### PR DESCRIPTION
**Pull request by merl-dev**
This will run on Julia 1.x, skipped about a dozen tests.  Some thought given to performance but nothing implemented here. Will likely continue to work on it this week.

### CODE REVIEW

**Comment by merl-dev** [comment]
- Most skipped tests fixed.
- Examples running and generated workbooks.  
- Documenter won't complete on error ```UndefVarError: include not defined```

**Comment by merl-dev** [comment]
- Note that it builds successfully on nightly, but not on julia 1.0.3.  I develop on 1.1. These errors seem to be due to commit 6516b85 where I removed patch code for ```OffsetArrays.no_offset_view``` and ```IdentityUnitRange``` which made it's way into Base.  The latter failure makes sense, the former doesn't.
- The reason for the OSX failure on nightly is something I've noted sporadically on linux: **tst_scale.jl eager-fixed parameter tests**.  I can't reproduce the error as it happens only rarely.

File: docs/.gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
 build/
 site/
-src/generated/*
+#src/generated/*  # this should be removed
```
Comment (Evizero) [inline docs/.gitignore:3]: why?

Comment (merl-dev) [reply docs/.gitignore:3]: I have a bunch of items that will be removed when the upgrade is complete

Comment (Evizero) [reply docs/.gitignore:3]: the thing is this causes a couple of files to be commited that shouldn't be

Comment (merl-dev) [reply docs/.gitignore:3]: Yes I am aware ... have been holding back more edits, mostly doc related.

**Comment by simonschoelly** [comment]
@merl-dev Are you still working on this PR?

**Comment by merl-dev** [comment]
> @merl-dev Are you still working on this PR?

Not presently

**Comment by johnnychen94** [comment]
I'm starting to continue the update work #27 and here in #31 , so I guess this PR is deprecated now and only serves for reference usage, will be closed when it successfully updates to 1.x

**Comment by johnnychen94** [comment]
close this by #31 

Although there're some bugs need to be fixed, branch jc/1.x is now 1.x-compatible

**Pull request closed without merging**